core/lib/via_btc_client/src/indexer/mod.rs

```diff
@@ -1 +1,243 @@
+use std::collections::HashMap;
 
+use async_trait::async_trait;
+use bitcoin::{Address, BlockHash, KnownHrp, Network, Txid};
+use bitcoincore_rpc::Auth;
+
+mod parser;
+use parser::MessageParser;
+
+use crate::{
+    client::BitcoinClient,
+    traits::BitcoinInscriptionIndexerOpt,
+    types::{
+        BitcoinError, BitcoinIndexerResult, CommonFields, FullInscriptionMessage, L1ToL2Message,
+        Vote,
+    },
+    BitcoinOps,
+};
+
+struct BootstrapState {
+    verifier_addresses: Vec<Address>,
+    proposed_sequencer: Option<Address>,
+    proposed_sequencer_txid: Option<Txid>,
+    sequencer_votes: HashMap<Address, Vote>,
+    bridge_address: Option<Address>,
+    starting_block_number: u32,
+}
+
+impl BootstrapState {
+    fn new() -> Self {
+        Self {
+            verifier_addresses: Vec::new(),
+            proposed_sequencer: None,
+            proposed_sequencer_txid: None,
+            sequencer_votes: HashMap::new(),
+            bridge_address: None,
+            starting_block_number: 0,
+        }
+    }
+
+    fn is_complete(&self) -> bool {
+        !self.verifier_addresses.is_empty()
+            && self.proposed_sequencer.is_some()
+            && self.bridge_address.is_some()
+            && self.starting_block_number > 0
+            && self.has_majority_votes()
+    }
+
+    fn has_majority_votes(&self) -> bool {
+        let total_votes = self.sequencer_votes.len();
```

Comment:
in here we should check that the number of vote is equal with number of verifiers
otherwise the voting is not ended yet

Reply:
tbh I don't really like the way to operate votes here, I think it should be moved
but let me change it for now

Reply:
I agree, and also i think in the btc_watcher we may need the tx context (including reveal,commit  txid, consumed fee, ...)

but let's keep it like this and fix them during the implementation of the btc_watcher

Reply:
btw, LGTM merge it after fixing this issue